plugins/micromega/zify.ml

```diff
@@ -1596,8 +1607,10 @@ let saturate =
           Array.iter sat args;
           if Array.length args = 2 then
             let ds = get_all_sat env evd c in
-            if ds = [] then ()
-            else List.iter (fun x -> CstrTable.HConstr.add table t x.deriv) ds
+            if ds = [] ||  CstrTable.HConstr.mem table t
```

Comment:
Why is the check `mem table t` done after computing `ds`?

Reply:
Both check are sufficient conditions for doing nothing. Both tables are expected to be very small.
The `get_all_sat` is indexed by `GlobRef.t` whereas `CstrTable.HConstr.mem` may work with bigger terms.
I also expect `ds` to be [] most of the time. 
PS: I did not benchmark